cmd/spire-server/cli/agent/list.go

```diff
@@ -40,20 +40,38 @@ func (listCommand) Synopsis() string {
 // Run lists attested agents
 func (c *listCommand) Run(ctx context.Context, env *common_cli.Env, serverClient util.ServerClient) error {
 	agentClient := serverClient.NewAgentClient()
-	listResponse, err := agentClient.ListAgents(ctx, &agentv1.ListAgentsRequest{})
-	if err != nil {
-		return err
+	pageToken := ""
+	moreResults := true
+	numResponseAgents := 0
+	var responseAgents []*types.Agent
+	var listResponse *agentv1.ListAgentsResponse
+	var err error
+
+	for moreResults {
+		listResponse, err = agentClient.ListAgents(ctx, &agentv1.ListAgentsRequest{
+			PageSize:  1396, // ~80% of (4 MB/theoretical maximum size of 1 agent in MB)
```

Comment:
Thank you for the comment explaining the number. The "back of the napkin math" seems like a reasonable default.

Perhaps a future enhancement could be addition of an explicit `--pagesize` option to the `list` command to override this default, but since the pagination is looped through prior to output, this is essentially hidden from the user so not needed.

It looks like server will only override this if server is configured with a smaller page size: https://sourcegraph.com/github.com/spiffe/spire-api-sdk/-/blob/proto/spire/api/server/agent/v1/agent.pb.go?L125-127&subtree=true